internals-js/src/federation.ts

```diff
@@ -1827,6 +1827,13 @@ class ExternalTester {
   }
 
   private collectExternalsOnType() {
+    // We do not collect @external on types for fed1 schema as those are ignored and will be discarded by the schema upgrader
+    // Do note however that the schema upgrader relying on the methods of this tester to figure out if a subgraph resolves
+    // a field in order to know when @shareable should be automatically added, hence the need for the short-cut here.
+    if (!this.isFed2Schema) {
+      return;
+    }
```

Comment:
For my own understanding I've rewritten this comment a bit. Is this correct, and if so would you consider using it here or updating yours a bit? Your PR description explains this but the comment misses this part which was important for me to understand it:
> In the fed1 case, if the map is populated then `@shareable` won't be added in places where it should have.

My interpretation / suggestion:
We do not collect `@external` on types for fed1 schema since those will be discarded by the schema upgrader.
The schema upgrader relies on the populated `externalFieldsOnType` object to inform when `@shareable` should be automatically added. In the fed1 case, if the map is populated then `@shareable` won't be added in places where it should have.

Reply:
The updates are correct, and are probably clearer; made the change.